internal/service/advancedclustertpf/resource.go

```diff
@@ -102,19 +102,26 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
 	if diags.HasError() {
 		return
 	}
+	if usingLegacySchema(ctx, plan.ReplicationSpecs, diags) && !usingLegacySchema(ctx, state.ReplicationSpecs, diags) {
+		diags.AddError("error operation not permitted, nums_shards from 1 -> > 1", fmt.Sprintf("cannot increase num_shards to > 1 under the current configuration. New shards can be defined by adding new replication spec objects; %s", DeprecationOldSchemaAction))
+		return
+	}
 	stateReq := normalizeFromTFModel(ctx, &state, diags, false)
 	planReq := normalizeFromTFModel(ctx, &plan, diags, false)
 	if diags.HasError() {
 		return
 	}
-	normalizePatchState(stateReq)
-	patchReq, err := update.PatchPayload(stateReq, planReq)
+	patchOptions := update.PatchOptions{
+		IgnoreInStatePrefix:  []string{"regionConfigs"},
+		IncludeInStateSuffix: []string{"diskIOPS"},
+	}
+	patchReq, err := update.PatchPayload(stateReq, planReq, patchOptions)
 	if err != nil {
 		diags.AddError("errorPatchPayload", err.Error())
 		return
 	}
 	var clusterResp *admin.ClusterDescription20240805
-	if !update.IsZeroValues(patchReq) {
+	if !update.IsZeroValues(patchReq) || findNumShardsUpdates(ctx, &state, &plan, diags) != nil {
```

Comment:
oh, why is this change necessary, isn't it detected in update?

Reply:
`num_shards` is only part of the legacy schema

Reply:
Updated again in 63c538e7bf7580a3c8a13e2e401c15bc9625eeab

Reply:
And again 2c0dbde44253bf246efacab595811d973bde69bb